Mage.Sets/src/mage/cards/a/AlaundoTheSeer.java

```diff
@@ -0,0 +1,235 @@
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SHAMAN);
+
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(5);
+
+        Ability ability = new SimpleActivatedAbility(new AlaundoTheSeerEffect(), new TapSourceCost());
+        this.addAbility(ability);
+    }
+
+    private AlaundoTheSeer(final AlaundoTheSeer card) {
+        super(card);
+    }
+
+    @Override
+    public AlaundoTheSeer copy() {
+        return new AlaundoTheSeer(this);
+    }
+}
+
+class AlaundoTheSeerEffect extends OneShotEffect {
+
+    private static final FilterCard inExileFilter = new FilterCard();
+    static {
+        inExileFilter.add(TargetController.YOU.getOwnerPredicate());
+        inExileFilter.add(CounterType.TIME.getPredicate());
+    }
+
+    public AlaundoTheSeerEffect() {
+        super(Outcome.PutCardInPlay);
+        this.staticText = "Draw a card, then exile a card from your hand and put a number of time "
+                + "counters on it equal to its mana value. It gains \"When the last time counter "
+                + "is removed from this card, if it's exiled, you may cast it without paying its "
+                + "mana cost. If you cast a creature spell this way, it gains haste until end of "
+                + "turn.\" Then remove a time counter from each other card you own in exile.";
+    }
+
+    private AlaundoTheSeerEffect(final AlaundoTheSeerEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public AlaundoTheSeerEffect copy() {
+        return new AlaundoTheSeerEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        // Draw a card
+        controller.drawCards(1, source, game);
+        if (controller.getHand().isEmpty()) {
+            return false;
+        }
+
+        // then exile a card from your hand and put a number of time counters on it equal to its mana value
+        TargetCard target = new TargetCardInHand().withChooseHint("to exile");
+        controller.chooseTarget(outcome, controller.getHand(), target, source, game);
+        Card card = game.getCard(target.getFirstTarget());
+        if (card == null) {
+            return false;
+        }
+        if (!CardUtil.moveCardWithCounter(game, source, controller,
+                card, Zone.EXILED, CounterType.TIME.createInstance(card.getManaValue()))) {
+            return false;
+        }
+
+        // It gains "When the last time counter is removed from this card, if it's exiled, you may cast it without paying its mana cost. 
+        // If you cast a creature spell this way, it gains haste until end of turn."
+        game.addEffect(new PlayFromExileEffect(new MageObjectReference(card, game)), source);
```

Comment:
General code style in this project is avoid naming card-specific classes with generic common names. So this could be `AlaundoPlayFromExileEffect` instead, and same for the others.

However, I'm pretty sure this could be simplified to use common methods in `CardUtil` rather than using three separate classes to implement this effect.

Reply:
Sure, i'll update the effect name.

I did initially try to implement this with existing `CardUtil` features as well as the existing suspend implementations, however if I recall (it was a few months ago now 😂) I think the main issue was giving perms haste.

I'm sure it's possible with some changes to some of the `CardUtil` functions, I just didn't feel confident in meddling with such a core set of utils. This card also has some really crazy edge cases due to it's interaction with other time counter cards that once I'd tested them all I just let it be.